frontend/src/common/components/mentoringStepper/MentoringStepper/MentoringStepper.tsx

```diff
@@ -0,0 +1,183 @@
+import styled from '@emotion/styled';
+
+import tooltipIcon from '../../../assets/images/tooltip.svg';
+import {
+  MentoringReservationStatusTypeEnum,
+  type MentoringReservationStatusType,
+} from '../../../types/mentoringReservationStatus';
+import Step from '../Step/Step';
+
+interface MentoringStepperProps {
+  status: MentoringReservationStatusType;
+}
+
+const statusTextMap = Object.values(MentoringReservationStatusTypeEnum).reduce(
+  (acc, key) => {
+    switch (key) {
+      case 'PENDING':
+        acc[key] = '예약신청';
+        break;
+      case 'APPROVED':
+        acc[key] = '신청확정';
+        break;
+      case 'COMPLETE':
+        acc[key] = '멘토링완료';
+        break;
+    }
+    return acc;
+  },
+  {} as Record<string, string>,
+);
+
+const statusInfoTextMap: Record<MentoringReservationStatusTypeEnum, string> = {
+  PENDING: '예약 확정 완료 시 문자로 연락용 오픈카톡방 링크가 발송됩니다.',
+  APPROVED:
+    '신청이 확정되었습니다. 문자로 발송된 오픈카톡 링크를 통해 멘토와 대화를 시작할 수 있습니다.',
+  COMPLETE: '멘토링이 완료되었습니다.',
+};
+
+function MentoringStepper({ status }: MentoringStepperProps) {
+  const stepValues = Object.values(MentoringReservationStatusTypeEnum);
+
+  const stepOrder = stepValues.map((value, index) => ({
+    step: index,
+    status: value,
+  }));
+
+  const matchedStatus = stepOrder.find((item) => item.status === status);
+  const currentStep = matchedStatus ? matchedStatus.step : 0;
+
+  const getType = (step: number) => {
+    if (step > currentStep) {
+      return 'after';
+    }
+
+    if (step === currentStep) {
+      return 'current';
+    }
+
+    return 'before';
+  };
+
+  return (
+    <StyledContainer>
+      <StyledSteps>
+        {stepOrder.map((stepInfo) => (
+          <>
+            <Step
+              type={getType(stepInfo.step)}
+              status={status}
+              key={stepInfo.status}
+            >
+              <StyledTextWrapper>
+                <StyledText step={getType(stepInfo.step)}>
+                  {statusTextMap[stepInfo.status]}
+                </StyledText>
+                <StyledIconWrapper>
+                  <StyledIcon src={tooltipIcon} alt="툴팁 아이콘" />
+                  <StyledTooltip>
+                    {statusInfoTextMap[stepInfo.status]}
```

Comment:
p4: 위에서 이어지는 내용인데, stepValues로 배열을 만든다면 이렇게 할 수도 있을 것 같아!

```suggestion
 {stepValues.map((stepInfo, step) => (
          <>
            <Step type={getType(step)} status={status} key={stepInfo}>
              <StyledTextWrapper>
                <StyledText step={getType(step)}>
                  {statusTextMap[stepInfo]}
                </StyledText>
                <StyledIconWrapper>
                  <StyledIcon src={tooltipIcon} alt="툴팁 아이콘" />
                  <StyledTooltip>{statusInfoTextMap[stepInfo]}
```

Reply:
굳이 객체로 안바꾸고 이렇게 하는게 더 좋을 거 같아👍

Reply:
위의 커밋 확인하면 될거같아!!